src/app/application-dialog/application-dialog.component.ts

```diff
@@ -271,22 +281,20 @@ export class ApplicationDialogComponent implements OnInit {
       });
     });
 
-    const sufficientMonthlyPaymentFormControls = [
-      this.realEstatePrice,
-      this.loanAmount,
-      this.loanTerm,
-    ];
-    sufficientMonthlyPaymentFormControls.forEach((control) => {
-      control.valueChanges.subscribe(() => {
-        this.updateSufficientMonthlyPayment();
-      });
-    });
     this.paymentScheduleType.valueChanges.subscribe(() => {
       this.updateSufficientMonthlyPayment();
     });
     this.euribor.valueChanges.subscribe(() => {
       this.updateSufficientMonthlyPayment();
     });
+    this.personalDetailsForm.valueChanges.subscribe(() => {
```

Comment:
I don't see usage of canProceedToIncomeDetails
Also it seems that  this.personalDetailsForm.valid could be used instead of new variable